pkg/sync/syncWorker.go

```diff
@@ -299,15 +306,25 @@ func (s *syncWorker) setupStream(
 			err,
 		)
 	}
-	return stream, nil
+	originatorStream := &originatorStream{nodeID: nodeID, stream: stream}
+	for _, row := range result {
+		if uint32(row.OriginatorNodeID) == nodeID {
+			lastEnvelope, err := envUtils.NewOriginatorEnvelopeFromBytes(row.OriginatorEnvelope)
+			if err != nil {
```

Comment:
Are there cases where we would actually want to continue here? I worry about an invalid `ClientEnvelope` blocking replication forever because it errors.

Maybe it's the originator's fault for not validating properly before putting it in the DB and we _should_ stop?

Just want to make sure we've considered all the failure modes.

Reply:
Yes, it's a tricky call. I think it's similar to the general programming question of whether we should assert things that should always be true. I'm a bit more in the camp of a simpler programming model where things fail if our assumptions are violated, and we fix them - there are also other places in the codebase that rely on the DB deserialization working (e.g. when a query is made) and it would be preferable not to mask them in our responses to clients.

> Maybe it's the originator's fault for not validating properly before putting it in the DB and we should stop? Would only be an issue if the validation rules changed.

Yes, I think the assumption is that the originator should have serialized it properly before writing to the DB. Note we're not doing any validation on the envelope here - we're just checking that it can be deserialized. So it shouldn't be a problem if the validation rules changed, but it could happen if we make a breaking proto change without preparing it properly.

Reply:
Maybe the safest thing is to leave it as-is and if we have nodes that get jammed up we know we have a problem to fix.

> Note we're not doing any validation on the envelope here - we're just checking that it can be deserialized

FWIW, we do a little bit of validation on the `ClientEnvelope` to make sure the topic matches the payload type. But that all should be checked before it ever goes in the DB.